libs/live-config/src/FeatureFlags/firebaseFeatureFlags.ts

```diff
@@ -0,0 +1,110 @@
+import { PropsWithChildren } from "react";
+import { snakeCase } from "lodash";
+import semver from "semver";
+import { Feature, FeatureId } from "@ledgerhq/types-live";
+import { getEnv } from "@ledgerhq/live-env";
+import { LiveConfig } from "./LiveConfig";
+
+export type FirebaseFeatureFlagsProviderProps = PropsWithChildren<unknown>;
+export const formatToFirebaseFeatureId = (id: string) => `feature_${snakeCase(id)}`;
+
+export const checkFeatureFlagVersion = (feature: Feature) => {
+  const platform = LiveConfig.getInstance().platform;
+  if (feature && feature.enabled && platform) {
+    let featureSpecificVersion: string | undefined;
+    if (platform === "desktop") {
+      featureSpecificVersion = feature.desktop_version;
+    }
+    if (["ios", "android"].includes(platform)) {
+      featureSpecificVersion = feature.mobile_version;
+    }
+    if (
+      featureSpecificVersion &&
+      !semver.satisfies(LiveConfig.getInstance().appVersion, featureSpecificVersion, {
+        includePrerelease: true,
+      })
+    ) {
+      return {
+        enabledOverriddenForCurrentVersion: true,
+        ...feature,
+        enabled: false,
+      };
+    }
+  }
+  return feature;
+};
+
+export const isFeature = (key: string): boolean => {
+  if (!LiveConfig.getInstance().providerGetvalueMethod) {
```

Comment:
if we are not able to connect to the firebase remote server for any reason. we just want to disable the feature instead of raising an error.
Same logic in today's codebase : https://github.com/LedgerHQ/ledger-live/blob/b729005584b81c332edd56606a52064efda55c7d/apps/ledger-live-desktop/src/renderer/components/FirebaseFeatureFlags.tsx#L55